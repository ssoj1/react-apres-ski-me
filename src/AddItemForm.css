.card {
    background-color: #101010d1;
    color: #f0f2f5;
    border: 20px solid rgba(16, 16, 16, 0.63);
    box-shadow: inset 0 0 0 2px #ffffff;
    box-sizing: border-box;
}


.NewAddItemForm {
    padding: 3vw;
}

legend {
    font-size: medium;
}

label {
    font-size: medium;
}

.NewAddItemForm-addBtn {
    margin-top: 2vw;
    color: black;
    background-color: white;
}