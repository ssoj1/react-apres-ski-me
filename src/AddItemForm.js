import React, { useState } from "react";
import { Button, Form, FormGroup, Label, Input, Card, CardBody } from 'reactstrap';
import "./AddItemForm.css";

const defaultInitialFormData = {
    type: "snacks",
    name: "",
    description: "",
    recipe: "",
    serve: "",
};

/** Form for adding a menu item.
 * 
 * Props: 
 * - initialFormData
 * - addItem - a function run on the parent that adds an item to the API and 
 * snacks or drinks array, depending on type
 * 
 * State: 
 * - none
 * 
 * Routes -> AddItemForm
 */

function AddItemForm({ initialFormData = defaultInitialFormData, addItem }) {
    console.debug("* AddItemForm ", { initialFormData, addItem });

    const [formData, setFormData] = useState(initialFormData);

    /** Update form input. */
    function handleChange(evt) {
        const input = evt.target;
        setFormData(formData => ({
            ...formData,
            [input.name]: input.value,
        }));
    }

    /** Call parent function and clear form. */
    function handleSubmit(evt) {
        evt.preventDefault();
        console.log("Submitted form data -->", formData);

        addItem(formData);
        setFormData(initialFormData);
    }

    return (
        <Card>
            <CardBody>
                <Form className="NewAddItemForm" onSubmit={handleSubmit}>

                    <FormGroup>
                        <Label
                            for="NewAddItemForm-type">
                                Is this a Snack or Drink (pick one)?
                        </Label>
                        <Input
                            type="select"
                            name="type"
                            id="NewAddItemForm-type"
                            value={formData.type}
                            onChange={handleChange}
                        >
                            <option value="snacks">Snack</option>
                            <option value="drinks">Drink</option>
                        </Input>
                    </FormGroup>

                    <FormGroup>
                        <Label
                            for="NewAddItemForm-name">Name
                        </Label>
                        <Input
                            name="name"
                            id="NewAddItemForm-name"
                            onChange={handleChange}
                            value={formData.name}
                            required
                        />
                    </FormGroup>

                    <FormGroup>
                        <Label
                            for="NewAddItemForm-description">Description
                        </Label>

                        <Input
                            name="description"
                            id="NewAddItemForm-description"
                            onChange={handleChange}
                            value={formData.description}
                            required
                        />

                    </FormGroup>

                    <FormGroup>
                        <Label
                            for="NewAddItemForm-recipe">Recipe
                        </Label>

                        <Input
                            name="recipe"
                            id="NewAddItemForm-recipe"
                            onChange={handleChange}
                            value={formData.recipe}
                            required
                        />

                    </FormGroup>

                    <FormGroup>
                        <Label
                            for="NewAddItemForm-serve">Serve
                        </Label>

                        <Input
                            name="serve"
                            id="NewAddItemForm-serve"
                            onChange={handleChange}
                            value={formData.serve}
                            required
                        />
                    </FormGroup>

                    <Button className="NewAddItemForm-addBtn">
                        Add!
                    </Button>

                </Form>
            </CardBody>
        </Card>
    );
}

export default AddItemForm;