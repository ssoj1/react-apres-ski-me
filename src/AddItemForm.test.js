import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import AddItemForm from "./AddItemForm";

test("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
          <AddItemForm />
      </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});