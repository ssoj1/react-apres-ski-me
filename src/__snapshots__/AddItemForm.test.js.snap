// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <form
        class="NewAddItemForm"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="NewAddItemForm-type"
          >
            Is this a Snack or Drink (pick one)?
          </label>
          <select
            class="form-control"
            id="NewAddItemForm-type"
            name="type"
          >
            <option
              value="snacks"
            >
              Snack
            </option>
            <option
              value="drinks"
            >
              Drink
            </option>
          </select>
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="NewAddItemForm-name"
          >
            Name
          </label>
          <input
            class="form-control"
            id="NewAddItemForm-name"
            name="name"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="NewAddItemForm-description"
          >
            Description
          </label>
          <input
            class="form-control"
            id="NewAddItemForm-description"
            name="description"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="NewAddItemForm-recipe"
          >
            Recipe
          </label>
          <input
            class="form-control"
            id="NewAddItemForm-recipe"
            name="recipe"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="NewAddItemForm-serve"
          >
            Serve
          </label>
          <input
            class="form-control"
            id="NewAddItemForm-serve"
            name="serve"
            required=""
            type="text"
            value=""
          />
        </div>
        <button
          class="NewAddItemForm-addBtn btn btn-secondary"
        >
          Add!
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
